cmdlets/src/main/java/org/corfudb/metrics/MBeanUtils.java

```diff
@@ -0,0 +1,152 @@
+                    String[] attributeNames = new String[beanInfoAttributes.length];
+                    for (int i = 0; i < beanInfoAttributes.length; i++) {
+                        attributeNames[i] = beanInfoAttributes[i].getName();
+                    }
+                    Arrays.sort(attributeNames);
+
+                    attributeList = mBeanServerConnection.getAttributes(objectName, attributeNames);
+                } catch (InstanceNotFoundException |
+                        ReflectionException |
+                        IOException e) {
+                    log.warn("Unable to fetch attributes for {}", objectName.toString());
+                }
+                jmxDump.put(objectName, attributeList);
+            }
+        }
+
+        return jmxDump;
+    }
+
+    /**
+     * Fetches the MBean object names from a JMX server. It filters the results
+     * by retrieving attributed for the provided domains
+     *
+     * @param mBeanServerConnection an open connection to a server.
+     * @param domains An array of domains for which attributes will be fetched from the
+     *                JMX server. An empty array result in returning object names for all
+     *                the domains.
+     * @return A set of MBean ObjectNames
+     *
+     * @throws IOException
+     * @throws MalformedObjectNameException
+     */
+    private static Set<ObjectName> getObjectNamesForDomains(@NonNull MBeanServerConnection mBeanServerConnection,
+                                                            @NonNull String[] domains)
+            throws IOException, MalformedObjectNameException {
+
+        // In case there are no domains, ObjectNames of all MBeans will be returned
+        if (domains.length == 0) {
+            return mBeanServerConnection.queryNames(null, null);
+        }
+
+        // In case domains are provided, all the ObjectNames within the provided domain will be returned
+        final Set<ObjectName> objectNames = new HashSet<>();
+        for (String domain : domains) {
+            final ObjectName objectNameFilter = new ObjectName(domain + ":*");
+            objectNames.addAll(mBeanServerConnection.queryNames(objectNameFilter, null));
+        }
+        return objectNames;
+    }
+
+    /**
+     * A convenience method for exporting a map of MBean object names and their attributes to
+     * a file.
+     *
+     * @param jmxDump A map of MBean object names along with their attributes.
+     * @param dumpFile A file name which along with operation timestamp will be
+     *                 used as the target dump file name.
+     *
+     * @throws FileNotFoundException
+     */
+    static void dumpMBeanAttributes(@NonNull Map<ObjectName, AttributeList> jmxDump,
+                                    @NonNull String dumpFile)
+            throws FileNotFoundException {
+
+        final File file = new File(dumpFile + System.currentTimeMillis());
+        final PrintStream printStream = new PrintStream(file);
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Use try-with-resources or close this "PrintStream" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2095)